### PR TITLE
MBS-11520 update AGP to 7.1

### PR DESCRIPTION
**Pull request by dsvoronin**

### CODE REVIEW

File: build-logic/publication/src/main/kotlin/convention.publish-android-library.gradle.kts [hunk 1]
```diff
@@ -17,13 +17,14 @@ configure<LibraryExtension> {
...
     var registeredVariants = 0
 
+    // todo use new publishing: https://developer.android.com/studio/releases/gradle-plugin#build-variant-publishing
     libraryVariants
         .matching {
             allVariantNames += it.name
             it.name == publishExtension.variant.get()
         }
         .whenObjectAdded {
-            publishing {
```
Comment (dsvoronin) [inline build-logic/publication/src/main/kotlin/convention.publish-android-library.gradle.kts:26]: android library now have it's own publishing extension: https://developer.android.com/studio/releases/gradle-plugin#build-variant-publishing

File: subprojects/delivery/sign-service/src/main/kotlin/com/avito/android/signer/SignServicePlugin.kt [hunk 2]
```diff
@@ -68,20 +62,4 @@ public class SignServicePlugin : Plugin<Project> {
             }
         }
     }
-
-    private fun checkSigningConfig(variant: ApplicationVariant) {
-        require(variant.signingConfig == null) {
```
Comment (dsvoronin) [inline subprojects/delivery/sign-service/src/main/kotlin/com/avito/android/signer/SignServicePlugin.kt:73]: signingConfig can't be null anymore, despite comment in javadoc (it's a mistake)

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 [versions]
```
Comment (eugene-krivobokov) [inline gradle/libs.versions.toml:1]: We need also to update r8

**Review by eugene-krivobokov — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 3 above)
Comment (dsvoronin) [reply gradle/libs.versions.toml:1]: removed a dependency, couldn't find usages

**Pull request merged**